cli/flox/doc-catalog/flox-search.md

```diff
@@ -0,0 +1,67 @@
+---
+title: FLOX-SEARCH
+section: 1
+header: "Flox User Manuals"
+...
+
+
+# NAME
+
+flox-search - search for packages
+
+# SYNOPSIS
+
+```
+flox [<general options>] search
+     [--json]
+     [-a]
+     <search-term>
+```
+
+# DESCRIPTION
+
+Search for available packages.
+
+Searches are performed in the context of the environment if one exists,
+making use of the environment's lock file and the locked base catalog within it
+if either one exists.
+Searches performed outside of an environment query a global base catalog.
+Both the global and environment's base catalogs can be updated with
+[`flox-update(1)`](./flox-update.md).
+
+A limited number of search results are reported by default for brevity.
+The full result set can be returned via the `-a` flag.
+
+Only the package name and description are shown by default.
+Structured search results can be returned via the `--json` flag.
+More specific information for a single package is available via the
+[`flox-show(1)`](./flox-show.md) command.
+
+```{.include}
+./include/package-names.md
+```
+
+## Fuzzy search
+`flox search` uses a fuzzy search mechanism that tries to match either the
+package name itself or some portion of the pkg-path.
```

Comment:
nonblocking/can do somewhere else: @billlevine are we searching description as well?

Reply:
We do a fuzzy match against the last tuple of the `attr_path` the `description`, and the `name`.  Currently the `name` is transitively the same as the `attr_path` though, and I don't think I would document that as part of the search at this time.

Reply:
Is it sufficient to say the following then?

> `flox search` uses a fuzzy search mechanism that tries to match either the
package name or package description.

On the basis that:

- we don't expose the term `attr_path` in the CLI
- "name" and `pkg-path` are anagolous at the moment
- the user might not be familiar with `pkg-path` if they start out with `flox search`

cc @jennymahmoudi @ghudgins

Reply:
I thought the `pkg-path` was the same as `attr_path` (right now), is what the user needs to specify for install, and is what is listed in the search results in the CLI.  Perhaps that's also referred to as `name` ?

Reply:
> I thought the pkg-path was the same as attr_path (right now), is what the user needs to specify for install, and is what is listed in the search results in the CLI.

Agree 👍🏻  The only caveat is that these docs are talking about the inputs to `flox search`, or how those inputs are used, rather than the outputs.

> Perhaps that's also referred to as name ?

What was originally referenced in these docs _might_ have been the first-class `name` from the [pre-catalog package descriptors](https://flox.dev/docs/reference/command-reference/manifest.toml/#package-descriptors) and isn't in the [post-catalog package descriptors](https://github.com/flox/flox/blob/main/cli/flox/doc-catalog/manifest.toml.md#package-descriptors). But maybe the word "name" is too loaded now? Is it any clearer to say some variation of the following? Wordsmithing welcome.

> flox search uses a fuzzy search mechanism that tries to match either pkg-path or package description.

Reply:
Ah, I see above:
> Only the package name and description are shown by default.

I feel that's confusing, but I think it's a matter on consistency throughout the man pages as to what we are referring to that as.. what `search` shows in the results.  It is in fact doing a fuzzy search on what it shows.  What do we call those things in the broad scope of the man pages.  Technically it's the `pkg-path`, but....

Reply:
I think we should probably use pkg-path more consistently

Reply:
I don't want to block these changes any further on the wording because it's already been a few days and several merge conflicts, so I've gone with the following and we can change it later when reviewing the docs before the catalog release.

> `flox search` uses a fuzzy search mechanism that tries to match either some portion of the pkg-path or description.